lib/jekyll/post.rb

```diff
@@ -155,6 +155,9 @@ def relative_path
     #
     # Returns -1, 0, 1
     def <=>(other)
+      return unless other.respond_to?(:date) &&
```

Comment:
Looking into it further, it seems like there's a bug that was being masked in earlier versions of Ruby. Could you take a look at the pull request conversation? I have a fix, I'm just not sure how to put together a test for it.